### PR TITLE
Adding payload arg as a templated field for Synapse SparkBatchOperator

### PR DESCRIPTION
**Pull request by narayave**
This change adds the "payload" argument as a templated field. This helps to send airflow macros as part of the payload and have them render before the rest api call is made.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/microsoft/azure/operators/synapse.py [hunk 1]
```diff
@@ -49,6 +49,7 @@ class AzureSynapseRunSparkBatchOperator(BaseOperator):
     template_fields: Sequence[str] = (
         "azure_synapse_conn_id",
         "spark_pool",
+        "payload"
```
Comment (Taragolis) [inline airflow/providers/microsoft/azure/operators/synapse.py:52]: Just wondering how it supposed to be work? According to the `payload` annotation it is expected to get `azure.synapse.spark.models.SparkBatchJobOptions` object

https://github.com/apache/airflow/blob/9d8c77e447f5515b9a6aa85fa72511a86a128c28/airflow/providers/microsoft/azure/operators/synapse.py#L63

For allow render templates for custom types it is required to have templated_field class method, see: [Jinja Templating](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/operators.html#jinja-templating)

**Review by hussein-awala — COMMENTED** [review]
I have the same concern as @Taragolis, would you add a new test case to check if your change works as expected?

File: airflow/providers/microsoft/azure/operators/synapse.py (hunk 1 above)
Comment (phanikumv) [inline airflow/providers/microsoft/azure/operators/synapse.py:52]: can we add an example DAG/docs for how this templated field would work

Comment (narayave) [reply airflow/providers/microsoft/azure/operators/synapse.py:52]: I would essentially look something like this. The templating allows to pass execution and task instance related attributes to the job.
In the following example I pass the logical execution datetime to the job using `{{ ts }}`.
```
    payload_example = {
        "name": "synapse_spark_def",
        "file": "abfss://<container>@<azstorageaccount>.dfs.core.windows.net/entrypoints/<filename>.py",
        "args": [
            "--exec_time",
            "{{ ts }}",    # Trying to pass something like a timestamp to job is where rendering will happen
        ],
        "jars": [],
        "pyFiles": [],
        "files": [],
        "conf": {
            "spark.dynamicAllocation.enabled": "false",
            "spark.dynamicAllocation.minExecutors": "3",
            "spark.dynamicAllocation.maxExecutors": "4",
        },
        "numExecutors": 3,
        "executorCores": 4,
        "executorMemory": "28g",
        "driverCores": 4,
        "driverMemory": "28g"
    }
```

**Comment by narayave** [comment]
I'm a first-time contributor could I get help with creating a unit test?

**Pull request closed without merging**